File: LogicalThinkingPatterns/AlphaHillPattern/AlphaHill.ts
class AlphaHill {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = n - i - 1; j > 0; j--) {
        pattern += "  ";
      }
      let ch: number = 65;
      for (let k = 0; k <= i; k++) {
        pattern += String.fromCharCode(ch) + " ";
        ++ch;
      }

      ch -= 2;
      for (let l = 0; l < i; l++) {
        pattern += String.fromCharCode(ch) + " ";
        --ch;
      }
      console.log(pattern);
    }
  }
}

const alphaHill = new AlphaHill();
alphaHill.printPattern(5);

File: LogicalThinkingPatterns/AlphaHillPattern/AlphaHill.py
class AlphaHill:
    def printPattern(self, n):
        for i in range(0, n):
            for j in range(n - i - 1, 0, -1):
                print("  ", end ="")
            
            ch = 'A'
            for k in range(0, i+1):
                print(ch+" ", end="")
                ch = chr(ord(ch)+1)
            
            ch = chr(ord(ch) - 2)
            for l in range(0, i):
                print(ch+" ", end="")
                ch = chr(ord(ch) - 1)
            
            print()
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

alphaHill = AlphaHill()
alphaHill.run()